main_test.go

```diff
@@ -614,3 +620,46 @@ func TestSubscribeCollationWithRPCNotifier(t *testing.T) {
 		t.Error("timeout in node2")
 	}
 }
+
+func TestRequestCollationWithEventNotifier(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	nodes := makeNodes(t, ctx, 2)
+	connectBarbell(t, ctx, nodes)
+	shardID := ShardIDType(1)
+	period := 2
+	collation, err := nodes[0].requestCollation(ctx, nodes[1].ID(), shardID, period)
+	if err != nil {
+		t.Errorf("request collation failed: %v", err)
+	}
+
```

Comment:
We can move the `if collation.ShardID != shardID || int(collation.Period) != period {}` below up here.

Reply:
Thanks a lot for pointing it out. I added cases and comments